### DISCUSSION TITLE
dummy package which contains some neutral value for all fields

### DISCUSSION THREAD

**Discussion by goyalyashpal** [ScoopInstaller/Scoop#5849] [Help and Questions] [answered]
hi

is there some dummy package which contains value in all the fields? even if it means just comments for the functional fields.

- it will come in handy a lot when trying my `jq` queries. 
- as often time i am typing all those from memory from quick recap, a typo sneaks somewhere here and there.
- but it just returns "null" in output; so, i have no way to test if the package lacks any value for that field or if i had put a wrong key.

```console
$ scoop cat python | jq '{uninstall}'
{
  "uninstall": null
}

```

so, having such a dummy package would help immediately spot such mistakes.

**Comment by niheaven** [comment, marked as answer]
Please check https://github.com/ScoopInstaller/Scoop/blob/master/schema.json for scoop manifest's Json Schema.

**Reply by goyalyashpal** [reply]
hi! i was aware that schema exists, but 
* schema can't be used directly on CLI to kinda semi-automatically verify the queries
* the schema can only be used to manually refer it for reading, or used in an elaborate tooling; not the plain CLI



umm, ohkay, i guess i can work with it nonetheless...

```bash
scoop alias add cat-dum 'curl.exe --silent -L github.com/ScoopInstaller/Scoop/raw/master/schema.json | jq.exe .properties' 'Fetches the portion of schema usable as dummy scoop cat'
```

```console
$ scoop cat-dum | head
{
  "$schema": {
    "type": "string",
    "format": "uri"
  },
  "_comment": {
    "description": "Deprecated. Use ## instead.",
    "$ref": "#/definitions/stringOrArrayOfStrings"
  },
  "##": {

```

* the alias name `cat-dum` is picked so as to facilitate toolable navigation and quick interchange with `cat <app>`
* the better name obviously would have been `schema` or `schema-prop` but like i said, that has more steps on interchanging from `cat <app>
* i guess it won't work as a test for deeper level objects... but it should work for top-level objects?!

can u think of any downside to this approach?
thanks anyhow 😃 


<details>
<summary>list of `keys` used currently. just for reference</summary>

```console
$ scoop cat-dum | jq 'keys_unsorted'
```

```json
[
  "$schema",
  "_comment",
  "##",
  "architecture",
  "autoupdate",
  "bin",
  "persist",
  "checkver",
  "cookie",
  "depends",
  "description",
  "env_add_path",
  "env_set",
  "extract_dir",
  "extract_to",
  "hash",
  "homepage",
  "innosetup",
  "installer",
  "license",
  "msi",
  "notes",
  "post_install",
  "post_uninstall",
  "pre_install",
  "pre_uninstall",
  "psmodule",
  "shortcuts",
  "suggest",
  "uninstaller",
  "url",
  "version"
]

```

**Comment by goyalyashpal** [comment]
here it is:

https://github.com/ScoopInstaller/BucketTemplate/blob/master/bucket/app-name.json.template

(found via: https://github.com/ScoopInstaller/Scoop/discussions/5867#discussioncomment-9029924 )
